### PR TITLE
added tests for patch request by article Id [task 8]

### PR DESCRIPTION
**Pull request by rsdecoder**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -246,3 +246,66 @@ describe("POST", () => {
...
+          body: 'Bastet walks amongst us, and the cats are taking arms!',
+          created_at: '2020-08-03T13:14:00.000Z',
+          votes: 10,
+          article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+        }
+      ])
+    })
+
+  });
+  test('STATUS 404 should respond with an appropriate message when no content is sent with a patch request', () => {
```
Comment (ghost) [inline __tests__/app.test.js:280]: I'd argue this should be a bad request. An empty body is a failure of the user not that something is missing.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -246,3 +246,66 @@ describe("POST", () => {
...
+    return request(app)
+    .patch("/api/articles/3")
+    .send()
+    .expect(404)
+    .then((response)=> {
+    expect(response.body.msg).toBe("Not Found")
+    })
+  });
+  test('STATUS 404 should respond with an appropriate message when no content is sent with a patch request', () => {
```
Comment (ghost) [inline __tests__/app.test.js:289]: This should also be a bad request for the same reasons above

File: __tests__/app.test.js [hunk 3]
```diff
@@ -246,3 +246,66 @@ describe("POST", () => {
...
+    const votesToUpdate = {
+      inc_votes: 20
+    }
+    return request(app)
+    .patch("/api/articles/not-an-id")
+    .send(votesToUpdate)
+    .expect(400)
+    .then((response)=> {
+      expect(response.body.msg).toBe("Bad Request")
+    })
```
Comment (ghost) [inline __tests__/app.test.js:309]: I'd love to see a 404 for a valid article id that doesn't exist. And update your malformed body tests to the appropriate error code.

**Review by ghost — COMMENTED** [review]
A few parts to tidy up in your error handling here, but great effort.

File: __tests__/app.test.js (hunk 3 above)
Comment (rsdecoder) [reply __tests__/app.test.js:309]: I agree. I also missed a test for valid endpoint but article_id does not exist. I have added it now. Thank you very much for the comments.

**Pull request merged**